src/titiler/core/titiler/core/middleware.py

```diff
@@ -107,29 +109,41 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send):
         if scope["type"] != "http":
             return await self.app(scope, receive, send)
 
-        exception: Exception | None = None
-        try:
-            await self.app(scope, receive, send)
-        except Exception as e:
-            exception = e
-
         request = Request(scope, receive=receive)
-
         data = {
-            "method": request.method,
-            "referer": next(
+            "http.method": request.method,
+            "http.url": str(request.url),
+            "http.scheme": request.url.scheme,
+            "http.host": request.headers.get("host", request.url.hostname or "unknown"),
+            "http.target": request.url.path
+            + (f"?{request.url.query}" if request.url.query else ""),
+            "http.user_agent": request.headers.get("user-agent"),
+            "http.referer": next(
                 (request.headers.get(attr) for attr in ["referer", "referrer"]),
                 None,
             ),
-            "origin": request.headers.get("origin"),
-            "path": request.url.path,
-            "path_params": request.path_params,
-            "query_params": dict(request.query_params),
-            "headers": dict(request.headers),
+            "http.request_content_length": request.headers.get("content-length"),
+            "http.request.header.accept-encoding": request.headers.get(
+                "accept-encoding"
+            ),
+            "http.request.header.origin": request.headers.get("origin"),
+            "net.host.name": request.url.hostname,
+            "net.host.port": request.url.port,
+            "titiler.query_params": dict(request.query_params),
         }
 
+        telemetry.add_span_attributes(telemetry.flatten_dict(data))
```

Comment:
we need to add the span attributes `before` awaiting the app, otherwise these attributes will not get attached to any span because it will already be closed.